django_features/custom_fields/serializers.py

```diff
@@ -48,3 +52,94 @@ def get_choices(self, obj: CustomField) -> list:
         return CustomChoiceSerializer(
             CustomValue.objects.filter(field=obj), many=True
         ).data
+
+
+CustomFieldData = namedtuple(
+    "CustomFieldData",
+    ["id", "identifier", "choice_field", "multiple_choice"],
+)
+
+
+class CustomFieldBaseModelSerializer(serializers.ModelSerializer):
+    _custom_fields: list[CustomFieldData] = []
+
+    class Meta:
+        abstract = True
+        fields = "__all__"
+        model = None
+
+    def __init__(
+        self,
+        instance: Any = None,
+        data: Any = empty,
+        exclude_custom_fields: bool = False,
+        **kwargs: Any,
+    ) -> None:
+        self.exclude_custom_fields: bool = exclude_custom_fields
+        super().__init__(instance, data, **kwargs)
+
+    @property
+    def model(self) -> models.Model:
+        if not self.Meta.model:
+            raise ValueError("Meta.model must be set")
+        return self.Meta.model
+
+    def get_fields(self) -> dict[str, Any]:
+        fields = super().get_fields()
+        if self.exclude_custom_fields:
+            return fields
+        custom_fields = list(CustomField.objects.for_model(self.model))
+        for field in custom_fields:
+            self._custom_fields.append(
+                CustomFieldData(
+                    field.id,
+                    field.identifier,
+                    field.choice_field,
+                    field.multiple_choice,
+                )
+            )
+            if field.choice_field:
+                fields[f"{field.identifier}_id"] = ChoiceIdField(
+                    required=field.required, write_only=True
+                )
+            fields[field.identifier] = field.serializer_field
+        return fields
+
+    def create(self, validated_data: dict) -> Any:
+        custom_value_instances: list[CustomValue] = []
+        choices: list[CustomValue] = []
+        for field in self._custom_fields:
+            if not field.choice_field:
+                value = validated_data.pop(field.identifier, None)
+                if not value:
```

Comment:
```suggestion
                if value is None:
```
Else a number field with 0 would be skipped.  If empty strings and lists also should be skipped add them accordingly.

Reply:
Already done in https://github.com/4teamwork/django-features/pull/9